api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -83,6 +83,19 @@ def show
             render_serialized_payload serialized_current_order, 200
           end
 
+          def apply_coupon_code
+            spree_authorize! :update, spree_current_order, order_token
+
+            spree_current_order.coupon_code = params[:coupon_code]
+            result = dependencies[:coupon_handler].new(spree_current_order).apply
+
+            if result.error.blank?
+              render_serialized_payload serialized_current_order, 200
```

Comment:
What do you think about creating some generic method which will receive result as an argument & render payload or error based on it? I suppose this isn't the only place we're going to render 422, right?

Reply:
@kushniryb currently it's not that unified - I would do that outside of the scope of this PR - this requires more thought put into it

Reply:
Okay, makes sense, let's discuss it in the future